src/main/java/com/github/tomakehurst/wiremock/http/AbstractRequestHandler.java

```diff
@@ -41,15 +41,18 @@ protected void afterResponseSent(ServeEvent serveEvent, Response response) {}
 
 	@Override
 	public void handle(Request request, HttpResponder httpResponder) {
+		if (logRequests()) {
+			notifier().info("Request received:\n" + formatRequest(request));
```

Comment:
I'm not sure breaking this off into a separate log message is a good idea. If there are a number of concurrent requests arriving there's no guarantee your request and response would be in adjacent messages. Writing it all out in a single message ensures the response matches the request preceding it.